h3/src/client/stream.rs

```diff
@@ -162,6 +173,20 @@ where
         res
     }
 
+    /// Poll receive an optional set of trailers for the response.
+    pub fn poll_recv_trailers(
+        &mut self,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<Option<HeaderMap>, Error>> {
+        let res = self.inner.poll_recv_trailers(cx);
+        if let Poll::Ready(Err(e)) = &res {
```

Comment:
this code is redundant to the async `recv_trailers` version. I think it would be better to remove the `
    pub async fn recv_trailers(&mut self) -> Result<Option<HeaderMap>, Error> {` in `h3/src/connection.rs` and call this function within the `pub async fn recv_trailers` function. And the same on the server side.

This reduces the redundant error handling.

Reply:
Done.